Fix for potential security issue. Under certain circumstances assets …

raphaelbauer opened the pull request:
…controller did stream content from arbitrary directories (Christian B. / ra).

raphaelbauer committed:
Fix for potential security issue. Under certain circumstances assets controller did stream content from arbitrary directories (Christian B. / ra).

buildhive commented:
[Ninja Web Framework » ninja #634](https://buildhive.cloudbees.com/job/ninjaframework/job/ninja/634/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Ninja Web Framework » ninja #635](https://buildhive.cloudbees.com/job/ninjaframework/job/ninja/635/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

raphaelbauer committed:
Removed feature to serve static assets outside of the classpath. After discussion
              with the team we agreed that reading and serving files may open the door
              to a range of attack vectors. If someone wants to maintain a module to
              do so that's perfectly ok. But the framework itself will not serve anything
              from the file system - at least when the user can specify the file of what
              to serve (ra).

buildhive commented:
[Ninja Web Framework » ninja #636](https://buildhive.cloudbees.com/job/ninjaframework/job/ninja/636/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Ninja Web Framework » ninja #637](https://buildhive.cloudbees.com/job/ninjaframework/job/ninja/637/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

liudianpeng commented:
```
public Result createGender(Context context,
                          @JSR303Validation GenderDto genderDto,
                          Validation validation){
    if (validation.hasViolations()){
        context.getFlashScope().error(getErrorMessages(validation));
        context.getFlashScope().put("name", genderDto.getName());
        context.getFlashScope().put("code",genderDto.getCode());
        context.getFlashScope().put("pinyin",genderDto.getPinyin());
        context.getFlashScope().put("sequence",genderDto.getSequence());
        return Results.redirect("/sys/dict/gender/new");
    }else {
        Gender gender=new Gender();
        gender.setName(genderDto.getName());
        gender.setCode(genderDto.getCode());
        gender.setPinyin(genderDto.getPinyin());
        gender.setSequence(genderDto.getSequence());
        gender.setCreated(new Date());
        gender=genericDao.create(gender);
        return Results.redirect("/sys/dict/gender/" + gender.getId() + "/show");
    }
}
```

v

liudianpeng commented:
@raphaelbauer  hello ! I admire you very much.

liudianpeng commented:
I want to ask about ninja.return Results.redirect("/sys/dict/gender/" + gender.getId() + "/show");
